### ISSUE TITLE
Re-export websocket types

### ISSUE THREAD

**Issue by IlyaSemenov** [h3js/h3#716]
### Describe the feature

As a developer using websockets, I want to come up with generic helper functions working with websocket peers (such as for #715). However, `h3` doesn't re-export the type of peer, so I have to use introspection types:

```ts
type Peer = Parameters<NonNullable<Parameters<typeof defineWebSocketHandler>[0]["open"]>>[0]
```

Not only this is cumbersome, but also it loses "genericness" of the original type.

I propose to re-export `Peer` directly from `h3` (and some other involved types, e.g. `Message`, possibly others).

### Additional information

- [x] Would you be willing to help implement this feature?

**Comment by manniL** [comment]
Just get them from `crossws` as in

`import type { Peer, Message } from 'crossws'`

[Example repo ](https://github.com/manniL/alexander-lichter-websockets-nitro/blob/main/server/api/websocket.ts) + [video](https://www.youtube.com/watch?v=OfY7JcrqkPg)

**Comment by IlyaSemenov** [comment]
I believe this is a poor advice.

See, `crossws` is not a dependency of the user project, `h3` is. With pnpm, importing types from `crossws` is not available (and rightfully so).

If one adds `crossws` to the list of explicit dependencies, they end up with the burden of supporting compatible versions of `h3` and `crossws`, and may easily end up with multiple `crossws` versions, one being pulled by `h3` and another being pulled by the explicit project dependency.

The types used by the callbacks `h3` **itself** directly accepts should be defined by `h3` (possibly, but not necessarily, re-exported by whatever underlying library it uses).

**Comment by tobiasdiez** [comment]
I completely agree with the analysis of @IlyaSemenov. Adding `crossws` as a direct dependency to the library/project, just to get access to the types seems cumbersome and prone to version conflicts. @manniL could you please reopen this issue?

**Comment by ultimateshadsform** [comment]
I also would like to add many methods in crossws is missing. Also the ctx is `any` type which is not good.

![image](https://github.com/user-attachments/assets/b482f942-c594-4438-b075-ed93608f35af)

I would like to be able to kick a client if it doesn't respond to pings and currently there seem to be no such option. Or no documentation about.

**Comment by mnlght03** [comment]
+1 here

**Comment by rijkvanzanten** [comment]
Importing the types from the package directly rather than through `h3` also causes conflicts if the version in use by `h3` is different from the one in use in the project. At time of writing `crossws` is at `0.4.1`, but `h3` uses `0.3.5`. When importing `Peer` directly from the more recent `crossws` will start causing type issues like:

```
Argument of type 'import("/Users/rijkvanzanten/Developer/typography-interaction-2526/trivia/node_modules/.pnpm/crossws@0.3.5/node_modules/crossws/dist/index").Peer<import("/Users/rijkvanzanten/Developer/typography-interaction-2526/trivia/node_modules/.pnpm/crossws@0.3.5/node_modules/crossws/dist/index").AdapterInternal>' is not assignable to parameter of type 'import("/Users/rijkvanzanten/Developer/typography-interaction-2526/trivia/node_modules/.pnpm/crossws@0.4.1/node_modules/crossws/dist/index").Peer<import("/Users/rijkvanzanten/Developer/typography-interaction-2526/trivia/node_modules/.pnpm/crossws@0.4.1/node_modules/crossws/dist/index").AdapterInternal>'.
  Property '#private' in type 'Peer' refers to a different member that cannot be accessed from within type 'Peer'.ts(2345)
```
